tests/utility/test_opt_algo.py

```diff
@@ -0,0 +1,195 @@
+import pytest
+import numpy as np
+import random
+
+
+from src.quantum_gates.utilities import create_random_quantum_circuit, transpile_qiskit_circuit, setup_backend,fix_counts
+from src.quantum_gates.utilities import Optimizator
+from configuration.token import IBM_TOKEN, HUB, GROUP, PROJECT
+from src.quantum_gates.utilities import DeviceParameters
+from src.quantum_gates.simulators import MrAndersonSimulator
+from src.quantum_gates.circuits import BinaryCircuit
+from src.quantum_gates.gates import standard_gates
+from src.quantum_gates.backends import BinaryBackend
+from src.quantum_gates._simulation.simulator import _apply_gates_on_circuit
+
+backend_config = {
+    "hub": HUB,
+    "group": GROUP,
+    "project": PROJECT,
+    "device_name": "ibm_kyiv"
+}
+
+backend = setup_backend(IBM_TOKEN, **backend_config)
+
+location = "tests/helpers/device_parameters/ibm_kyiv/"
+
+
+def level_optimization(level: int, result: list, q: list, qc: list, n: int, psi0: np.array, sim: MrAndersonSimulator):
+    bb = BinaryBackend(n)
+    opt_ = Optimizator(level_opt=level, circ_list=result, qubit_list=q)
+    result_ = opt_.optimize()
+    psi_result_1 = bb.statevector(mp_list=result_ , psi0 = psi0,level_opt= 0, qubit_layout=q)
+    probs_ = np.square(np.absolute(psi_result_1))
```

Comment:
@paolodr98 What is the idea behind the underscores at the end of the variables?

Reply:
Because before defining the `level_optimization` function, these are 'free' variables with a numerical index referring to the optimization level. Then I create the function remove the numerical indexes, but I leave the underscore. I can remove it because there is no real meaning